projects/clr/rocclr/device/rocm/rocdevice.cpp

```diff
@@ -1987,6 +1987,7 @@ hsa_amd_memory_pool_t Device::getHostMemoryPool(MemorySegment mem_seg,
       segment = agentInfo->fine_grain_pool;
       break;
     case kUncachedAtomics:
+    case kIoMemory:
```

Comment:
Note:
On Navi4,  agentInfo->ext_fine_grain_pool.handle will be NULL but we still want hipHostRegisterIoMemory to use HSA_AMD_MEMORY_POOL_UNCACHED_FLAG instead of returning error.

I think there is another patch in progress where hipHostRegister will return error when agentInfo->ext_fine_grain_pool.handle is NULL. 
Assuming this patch is going in first, then it looks ok, but we may have to adjust the other patch.

Sorry, I have to step out, did not have time to look for the other patch. (I think Tao is working on the other change)